src/drivers/intel/cavs/ipc.c

```diff
@@ -238,6 +240,9 @@ void ipc_platform_ack_cmd(void)
 void ipc_platform_complete_cmd(struct ipc *ipc)
 {
 	ipc_msg_send_status();
+
+	if (ipc->pm_prepare_D3)
+		platform_pm_runtime_power_off();
```

Comment:
If you keep the send of the reply within the `ipc_cmd()` then this patch is not going to be needed...

Reply:
so you prefer to have a NOP version of ipc_platform_complete_cmd ?

Reply:
It is a good solution. @lgirdwood what is your idea ?

Reply:
Just to have less moving parts in where we put the platform to rest...

Reply:
I'm fine with @ujfalusi suggestion as long as we can enter the assembly for shutdown like we do today.